Library/PTPusherDelegate.h

```diff
@@ -111,41 +112,43 @@
 
 /** Allows the delegate to return authorization data in the format required by Pusher from a
  non-standard respnse.
- 
- When using a remote server to authorize access to a private channel, the server is expected to 
+
+ When using a remote server to authorize access to a private channel, the server is expected to
  return an authorization payload in a specific format which is then sent to Pusher when connecting
  to a private channel.
- 
+
  Sometimes, a server might return a non-standard response, for example, the auth data may be a sub-set
  of some bigger response.
- 
- If implemented, Pusher will call this method with the response data returned from the authorization
- URL and will use whatever dictionary is returned instead.
-*/
- - (NSDictionary *)pusher:(PTPusher *)pusher authorizationPayloadFromResponseData:(NSDictionary *)responseData;
+
+ By implementing this delegate method, you will be able to set any credentials as necessary by
+ modifying the request as required (such as setting POST parameters or headers).
+
+ See PTPusherChannelAuthorizationOperation for details on how to modify the outgoing request.
+ */
+- (void)pusher:(PTPusher *)pusher willAuthorizeChannel:(PTPusherChannel *)channel withAuthOperation:(PTPusherChannelAuthorizationOperation *)operation;
```

Comment:
Not sure this would be necessary anymore. If users need to customise channel auth beyond any API we expose on the basic implementation they can write their own really easily.